### PR TITLE
Add Counter and Mirror Coat Bugfix

### PR DESCRIPTION
**Pull request by hyperdriveguy**
A fix to check if the opponent's move did any damage. I've updated both the docs and added comments in the code.

### CODE REVIEW

**Comment by Rangi42** [comment]
Have you actually tested this fix? You put the `jr z, .capped` line right after `ld hl, wCurDamage`, before the contents of `wCurDamage` have even been inspected.

I think you meant to replace `ret z` with `jr z, .capped`.

**Comment by hyperdriveguy** [comment]
I swear I tested it, but you're probably right. I'll double check to see what works and update it.

**Comment by hyperdriveguy** [comment]
After extensive fiddling and testing, I think I've found the proper fix.

File: docs/bugs_and_glitches.md [hunk 1]
```diff
@@ -246,7 +246,34 @@ DefenseDownHit:
...
+	ld hl, wCurDamage
+	ld a, [hli]
+	or [hl]
+	jp z, .failed
+```
+
+Add this to the end of each file:
+
+```asm
+	.failed
```
Comment (mid-kid) [inline docs/bugs_and_glitches.md]: This shouldn't be indented. Otherwise these fixes look OK.

File: docs/bugs_and_glitches.md [hunk 2]
```diff
@@ -246,7 +246,34 @@ DefenseDownHit:
 
 ([Video](https://www.youtube.com/watch?v=uRYyzKRatFk))
 
-*To do:* Identify specific code causing this bug and fix it.
+This is a bug with `BattleCommand_Counter` in [engine/battle/move_effects/counter.asm](/engine/battle/move_effects/counter.asm) and `BattleCommand_MirrorCoat` in [engine/battle/move_effects/mirror_coat.asm](/engine/battle/move_effects/mirror_coat.asm):
+
+```asm
```
Comment (mid-kid) [inline docs/bugs_and_glitches.md]: Oh, and this should be in diff format.

File: docs/bugs_and_glitches.md (hunk 1 above)
Comment (hyperdriveguy) [reply docs/bugs_and_glitches.md]: Fixed in latest commit.

File: docs/bugs_and_glitches.md (hunk 2 above)
Comment (hyperdriveguy) [reply docs/bugs_and_glitches.md]: Changed it to the diff format. But I think it should be noted that none of the other bugfixes use the diff format.

**Comment by hyperdriveguy** [comment]
Ready to merge.

**Pull request merged**